app/controllers/v3/service_credential_bindings_controller.rb

```diff
@@ -328,14 +328,14 @@ def fetch_credentials_value(name)
   end
 
   def service_credential_binding
-    @service_credential_binding ||= fetcher.fetch(hashed_params[:guid], space_guids: space_guids)
+    @service_credential_binding ||= fetcher.fetch(hashed_params[:guid], readable_spaces_query: spaces_query)
   end
 
-  def space_guids
+  def spaces_query
     if permission_queryer.can_read_globally?
-      :all
+      nil
     else
-      permission_queryer.readable_space_guids
+      permission_queryer.readable_space_guids_query
```

Comment:
I would create a new method in `VCAP::CloudController::Permissions` named `readable_spaces_query` that uses `membership.authorized_spaces_subquery` instead of `membership.authorized_space_guids_subquery`.